crates/lovely-core/src/lib.rs

```diff
@@ -39,11 +39,13 @@ impl Lovely {
     pub fn init(loadbuffer: &'static LoadBuffer) -> Self {
         let start = Instant::now();
 
-        let args = std::env::args().skip(1).collect::<Vec<_>>();
-        let mut opts = Options::new(args.iter().map(String::as_str));
+        let mut it = std::env::args();
+        let arg1 = it.next().unwrap();
+        let args = it.collect::<Vec<_>>();
+        let game_name = Path::new(&arg1).file_stem().unwrap().to_string_lossy().replace(".", "_");        let mut opts = Options::new(args.iter().map(String::as_str));
```

Comment:
Refactor this block to something like 
```rust
let game_name = env::current_exe()
    .expect("Failed to get the path of the current executable.")
    .file_stem()
    .expect("Failed to get file_stem component of current executable path.")
    .to_string_lossy()
    .replace(".", "_");
  ```
and make sure to re-add the args var assignment
```rust
let args = std::env::args().skip(1).collect::<Vec<_>>();
```
This is more succinct and integrates some sort of error message so we know wtf is going on if we get a crash report from this bit of code. It *shouldn't* happen but you never know

Reply:
Changed